test/src/lib/update-lock.spec.ts

```diff
@@ -109,152 +105,168 @@ describe('lib/update-lock', () => {
 	});
 
 	describe('Lock/dispose functionality', () => {
-		const lockFile = updateLock.__get__('lockFile');
-		const locksTaken = updateLock.__get__('locksTaken');
-		const dispose = updateLock.__get__('dispose');
-		const lockExistsErrHandler = updateLock.__get__('lockExistsErrHandler');
+		const getLockParentDir = (): string =>
+			`${constants.rootMountPoint}${updateLock.lockPath(appId, serviceName)}`;
+
+		const expectLocks = async (exists: boolean = true) => {
+			expect(await fs.readdir(getLockParentDir())).to.deep.equal(
+				exists ? ['resin-updates.lock', 'updates.lock'] : [],
+			);
+		};
 
-		const releaseFn = stub();
-		const testLockPaths = ['/tmp/test/1', '/tmp/test/2'];
+		let unlockSpy: SinonSpy;
+		let lockSpy: SinonSpy;
+		let execStub: SinonStub;
 
-		let unlockSyncStub: SinonStub;
-		let unlockAsyncSpy: SinonSpy;
-		let lockAsyncSpy: SinonSpy;
+		let configGetStub: SinonStub;
 
 		beforeEach(() => {
+			unlockSpy = spy(lockfile, 'unlock');
+			lockSpy = spy(lockfile, 'lock');
+			// lockfile.lock calls exec to interface with the lockfile binary,
+			// so mock it here as we don't have access to the binary in the test env
 			// @ts-ignore
-			unlockSyncStub = stub(lockFile, 'unlockSync').callsFake((lockPath) => {
-				// Throw error on process.exit for one of the two lockpaths
-				if (lockPath === testLockPaths[1]) {
-					throw new Error(
-						'handled unlockSync error which should not crash test process',
-					);
-				}
+			execStub = stub(fsUtils, 'exec').callsFake(async (command, opts) => {
```

Comment:
I think it would be enough to stub the `lockfile` instead of stubbing `exec` here. The internals of the `lockfile` module have already been tested in the other file and here we are testing that this module responds well to that interface.

If we could test with the actual binary (e.g. within a container) we would not need any stubs and we could be sure that both modules behave as expected with a real filesystem, but that's a future improvement :)

Reply:
On second though, this might be too much of a nitpick, just think about whether it simplifies your tests.

Reply:
This sounds like a good improvement for when we move or add integration testing in an environment with the OS (balenaOS-in-container, eventually testbot), as discussed.